osmaxx-py/osmaxx/contrib/auth/frontend_permissions.py

```diff
@@ -45,3 +46,26 @@ class FrontendAccessRequiredMixin(object):
     @method_decorator(frontend_access_required)
     def dispatch(self, *args, **kwargs):
         return super(FrontendAccessRequiredMixin, self).dispatch(*args, **kwargs)
+
+
+class AuthenticatedAndAccessPermission(permissions.BasePermission):
+    """
+    Allows access only to authenticated users with frontend permissions.
+    """
+
+    def has_permission(self, request, view):
+        return request.user.is_authenticated() and _may_user_access_osmaxx_frontend(request.user)
+
+
+class HasBBoxAccessPermission(permissions.BasePermission):
+    message = 'Accessing this bounding box is not allowed.'
+
+    def has_object_permission(self, request, view, obj):
+        return obj.excerpt.is_public or obj.excerpt.owner == request.user
```

Comment:
I think I'd prefer to have the actual logic on the model (be it on the specific class or on a super class) and have a relatively 'dumb' method calling it (similar complexity to `_may_user_access_osmaxx_frontend`) in `frontend_permissions.py`. (Not placing the actual logic in `frontend_permissions.py` avoids 'Clean Code' smell G14. Somewhat similar to "Tell, don't ask.")

As our models are not intended to be used in other applications (that might have different access regimes), I don't think we'd obstruct anything by having this logic in the model.